Source/Core/VideoCommon/VertexLoaderManager.cpp

```diff
@@ -265,6 +267,82 @@ int RunVertices(int vtx_attr_group, OpcodeDecoder::Primitive primitive, int coun
   if (is_preprocess)
     return size;
 
+  // Validate that the XF input configuration matches the CP configuration
+  u32 num_cp_colors = std::count_if(
+      g_main_cp_state.vtx_desc.low.Color.begin(), g_main_cp_state.vtx_desc.low.Color.end(),
+      [](auto format) { return format != VertexComponentFormat::NotPresent; });
+  u32 num_cp_tex_coords = std::count_if(
+      g_main_cp_state.vtx_desc.high.TexCoord.begin(), g_main_cp_state.vtx_desc.high.TexCoord.end(),
+      [](auto format) { return format != VertexComponentFormat::NotPresent; });
+
+  u32 num_cp_normals;
+  if (g_main_cp_state.vtx_desc.low.Normal == VertexComponentFormat::NotPresent)
+    num_cp_normals = 0;
+  else if (g_main_cp_state.vtx_attr[vtx_attr_group].g0.NormalElements == NormalComponentCount::NTB)
+    num_cp_normals = 3;
+  else
+    num_cp_normals = 1;
+
+  u32 num_xf_normals;
+  switch (xfmem.invtxspec.numnormals)
+  {
+  case NormalCount::None:
+    num_xf_normals = 0;
+    break;
+  case NormalCount::Normal:
+    num_xf_normals = 1;
+    break;
+  case NormalCount::NormalTangentBinormal:
+    num_xf_normals = 3;
+    break;
+  default:
+    PanicAlertFmt("xfmem.invtxspec.numnormals is invalid: {}", xfmem.invtxspec.numnormals);
+    num_xf_normals = 888;
```

Comment:
Why 888?

Reply:
It's an arbitrary number that `num_cp_normals` cannot be (so that the second test always fails); leaving it uninitialized would result in undefined behavior, and using 0 could result in no game quirk being reported.

Reply:
I know the c++ community has different views on it.  But rather than magic numbers for values being undefined, I personally prefer `std::optional`.

Reply:
If you do want to use a magic number like this, please add a constant or a comment.

Reply:
I thought that `std::optional` would make the `num_cp_normals != num_xf_normals` check more complicated (instead requiring `!num_xf_normals || num_cp_normals != num_xf_normals` but it seems like `num_cp_normals` will be automatically converted to a present optional and if `num_xf_normals` is not present then [the two are considered not equal](https://en.cppreference.com/w/cpp/utility/optional/operator_cmp).  There isn't a `fmt::formatter` for `std::optional` though so that logic is more complicated too...

Reply:
Don't know if it'd be useful to do so but we could add a formatter for optional, someone did so here:  https://github.com/fmtlib/fmt/issues/1367

Reply:
Eh, I think specializing it is OK since it means we can use a specific message for the not-present case (rather than always using "NO VALUE" or the like).  If it ends up being a more common problem then we can add one though.